cpp/unreal_projects/SpearSim/Python/validate_scene_kujiale.py

```diff
@@ -0,0 +1,591 @@
+
+    actors = sorted(actors, key=(lambda actor: get_debug_string_actor(actor)))
+    print(len(actors))
+    for actor in actors:
+        validate_actor(level, actor)
+
+
+def validate_actor(level, actor):
+    #
+    # Actors that are useful for debugging are kept here.
+    #
+    # All Actors that represent non-trivial scene geometry are kept here.
+    #
+    # All navigation-related Actors are kept here. There must be at least one RecastNavMesh
+    # Actor in the scene in order to use the navmesh functionality in our Python API.
+    #
+    # All rendering-related Actors are kept here. For convenience, we allow content here to
+    # deviate from our usual naming conventions.
+    #
+    # All settings-related Actors are kept here.  For convenience, we allow content here to
+    # deviate from our usual naming conventions.
+    #
+    # All of our custom Widgets are kept here.
+    #
+
+    path = pathlib.PurePosixPath(str(actor.get_folder_path())).parts
+
+    # if len(path) < 1:
+    #     spear.log("WARNING: Unexpected path: ", get_debug_string_actor(actor))
+    #     return
+    #
+    # if path[0] not in ["Debug", "Meshes", "Navigation", "Rendering", "Settings", "Widgets"]:
+    #     spear.log("WARNING: Unexpected path: ", get_debug_string_actor(actor))
+    #     return
+    #
+    # if path[0] in ["Meshes", "Rendering"]:
+    #     if len(path) != 2:
+    #         spear.log("WARNING: Unexpected path: ", get_debug_string_actor(actor))
+    #         return
+    #
+    # if path[0] in ["Navigation", "Settings", "Widgets"]:
+    #     if len(path) != 1:
+    #         spear.log("WARNING: Unexpected path: ", get_debug_string_actor(actor))
+    #         return
+
+    if path[0] == "Meshes":
+
+        #
+        # We create a subdirectory for each semantic category using a lower_case_with_underscore
+        # naming convention and a four digit prefix that represents the category's integer ID.
+        # We assign each Actor to a particular semantic category by placing the Actor in that
+        # category's subdirectory. Semantic ID 0 is reserved and should not be used here. If an
+        # Actor has multiple StaticMeshComponents that should have different categories, the
+        # category of individual StaticMeshComponents can be overridden. In this case, the Actor
+        # should be assigned to whatever subdirectory is most convenient, i.e., requires the least
+        # number of per-component overrides.
+        #
+        # We choose to encode semantic annotations via this directory structure, because it
+        # makes it especially easy to browse each scene in the Unreal Editor. For example,
+        # using this directory structure, a user can easily select all objects in a particular
+        # semantic category and make them invisible, thereby making it easier to browse the
+        # rest of the scene.
+        #
+
+        semantic_category = path[1].split("_")
+
+        # if len(semantic_category) < 2:
```

Comment:
@rwang15 Why is this commented out? Don't we want to check for these conditions?

As an aside, we generally prefer to not to check in commented out code like this, unless it is serving some kind of useful purpose as documentation. Chunks of commented out code linger in codebases forever.